Test/git-issues/git-issue-921a.dfy

```diff
@@ -0,0 +1,10 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+module StringsOfLength {
+  type ShortString = s: string | |s| <= 2
+
+  method M() {
+    var x2: ShortString := "12" + "34"; // no error!
```

Comment:
The comment says "no error", but there is an error and it's being reported.

This test can also be moved into `git-issue-921.dfy`. I don't see a reason to have a separate file for it.

Reply:
fixed and combined the tests. (The second test file was a smaller case for debugging.)